api/redhatopenshift/HcpCluster.Management/hcpOperatorIdentityRoles.tsp

```diff
@@ -0,0 +1,82 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+namespace Microsoft.RedHatOpenShift;
+
+using Azure.ResourceManager;
+
+/** HcpOperatorIdentityRoles represents a location based representation of
+ * the required platform workload identities and their required roles for a given
+ * OpenShift version
+ */
+@locationResource
+model HcpOperatorIdentityRoleSet
+  is ProxyResource<HcpOperatorIdentityRoleSetProperties> {
+  ...ResourceNameParameter<
+    HcpOperatorIdentityRoleSet,
+    NamePattern = "^[a-zA-Z0-9-\\.]{3,24}$"
+  >;
+}
+@@doc(HcpOperatorIdentityRoleSet.name,
+  "The OpenShift minor version these identities are required for."
+);
+
+/** HCP Operator Identity Roles properties */
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "Proxy resource does not have provisioning state"
+model HcpOperatorIdentityRoleSetProperties {
+  /** The role definitions required for the User-Assigned managed identities used
+   * by Control Plane operators on a cluster. */
+  controlPlaneOperators: OperatorIdentityRoles[];
+
+  /** The role definitions required for the User-Assigned managed identities used
+   * by Data Plane operators on a cluster. */
+  dataPlaneOperators: OperatorIdentityRoles[];
+}
+
+/** Role definitions for a specific operator */
+model OperatorIdentityRoles {
```

Comment:
Is this missing min/max supported openshift version?

Or the info is not needed since the user will do something like `Give me the required identity roles for the X openshift version I am going to install?`
i.e
```
GET /subscriptions/<subscriptio	n-id>/locations/<location>/providers/Microsoft.RedHatOpenShift/hcpOperatorIdentityRoleSets/4.14.
```

? 

if so, how does the response to the query below look like 

```
GET /subscriptions/<subscriptio	n-id>/locations/<location>/providers/Microsoft.RedHatOpenShift/hcpOperatorIdentityRoleSets/4.14.
```
? or we won't be supporting that endpoint

Reply:
Actually, ignore what I said. The info I was asking is on the `.name` field of the returned response.